### PR TITLE
Update building exaple to allow streaming output to *out*

### PR DESCRIPTION
**Pull request by davidpham87**
It might add a bit of noise to the example, but I think it is helpful to have the output of the command print in the shell :-)

### CODE REVIEW

**Comment by lispyclouds** [comment]
Looks good! But I think if we made it into two separate examples would be nicer. People may not be interested in the build output always and keeping it specific is nicer.

**Comment by lispyclouds** [comment]
Maybe we can call the other one as processing the streaming build outputs and have a generic fn like `(process-build-output ...)` than just printing for example?

**Comment by davidpham87** [comment]
Yes, definitively! Sorry for not making it in the first pull request.

File: doc/002-general-guidelines.md [hunk 1]
```diff
@@ -52,18 +54,30 @@ script would build the `contajners-build-example` image.
...
+
+(defn build! [& {:keys [verbose?]}]
+  (let [docker-output-stream (build-cmd!)]
+    (when verbose?
+      (show-build-output! docker-output-stream))))
 
 (->tar!)
-(->build!)
+(->build! :verbose? true)
+;; (->build! {:verbose? true}) ;; in 1.11
```
Comment (lispyclouds) [inline doc/002-general-guidelines.md:80]: when would this comment be needed?

Comment (davidpham87) [reply doc/002-general-guidelines.md:80]: There is a new keyword syntax in Clojure 1.11. But we could remove it indeed.

Comment (lispyclouds) [reply doc/002-general-guidelines.md:80]: ah right, i didnt get the intention. maybe we can add clojure 1.11+ to make it a bit more explicit?

Comment (lispyclouds) [reply doc/002-general-guidelines.md:80]: I'll add it, thanks for this again!

**Pull request merged**